app/services/encryption/multi_region_kms_client.rb

```diff
@@ -0,0 +1,78 @@
+require 'json'
+module Encryption
+  class MultiRegionKMSClient
+    def initialize
+      @aws_clients = {}
+      # Instantiate an array of aws clients based on the provided regions in the environment
+      JSON.parse(Figaro.env.aws_kms_regions).each do |region|
+        @aws_clients[region] = Aws::KMS::Client.new(
+          instance_profile_credentials_timeout: 1, # defaults to 1 second
+          instance_profile_credentials_retries: 5, # defaults to 0 retries
+          region: region, # The region in which the client is being instantiated
+        )
+      end
+    end
+
+    # Use each KMS client to encrypt with params and options
+    # Region format example: '{"reg": {"us-east-1": "cipher", "us-west-2": "othercipher"}}'
+    def encrypt(key_id, plaintext, encryption_context)
+      region_ciphers = {}
+      @aws_clients.each do |region, kms_client|
+        r_cipher = kms_client.encrypt(key_id: key_id,
+                                      plaintext: plaintext,
+                                      encryption_context: encryption_context)
+        region_ciphers[region] = r_cipher.ciphertext_blob
+      end
+      { reg: region_ciphers }.to_json
+    end
+
+    def decrypt(ciphertext, encryption_context)
+      # The client that will be used to decode the ciphertext, the ciphertext after being parsed out
+      # of whatever format it's in
+      region_client, resolved_ciphertext = resolve_decryption(ciphertext)
+      # If the ciphertext is valid and there's a relevant client, try to actually decode it
+      region_client.decrypt(
+        ciphertext_blob: resolved_ciphertext,
+        encryption_context: encryption_context,
+      ).plaintext
+    end
+
+    private
+
+    def resolve_region_decryption(regions)
+      # For each region that the ciphertext has a cipher for, check to see if that region is
+      # represented in the clients available
+      regions.each do |region, cipher|
+        region_client = @aws_clients[region]
+        resolved_ciphertext = cipher
+        return region_client, resolved_ciphertext if region_client
+      end
+      raise EncryptionError, 'No supported region found in ciphertext'
+    end
+
+    def resolve_legacy_decryption(ciphertext)
+      # Decode the raw ciphertext
```

Comment:
I notice there are a lot of comments in this class -- I appreciate the commitment to documentation! However, I think that maybe for this one in particular, it seems simple enough to remove it.

A former coworker used to say "comments are lies waiting to happen" because often code changes and comments stay and get stale, so I try to minimize comments to keep them to they're absolutely necessary

Reply:
That's a great saying! I'll work on cutting down the verbosity then